apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -67,4 +116,98 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  def get_txs_for_address_with_proof(user_txs) do
+    blocks_for_user_txs =
+    for user_tx <- user_txs do
+      get_block_by_txs_hash(user_tx.txs_hash)
+    end
+    merkle_trees =
+    for block <- blocks_for_user_txs do
+      build_tx_tree(block.txs)
+    end
+    user_txs_trees = Enum.zip(user_txs, merkle_trees)
+    proof =
+    for user_tx_tree <- user_txs_trees  do
+      {tx, tree} = user_tx_tree
+      transaction = :erlang.term_to_binary(
+        tx
+        |> Map.delete(:txs_hash)
+        |> Map.delete(:block_hash)
+        |> Map.delete(:block_height)
+        |> Map.delete(:signature)
+        |> TxData.new()
+      )
+      key = :crypto.hash(:sha256, transaction)
```

Comment:
We talked about extracting the transaction hashing to `Aecore.Structures.TxData` module or some utility module, remember? ;)